src/token/ERC20Gauges.sol

```diff
@@ -121,7 +140,13 @@ abstract contract ERC20Gauges is ERC20, Auth {
     */
     function calculateGaugeAllocation(address gauge, uint256 quantity) external view returns(uint256) {
         if (!_gauges.contains(gauge)) return 0;
-        return quantity * getGaugeWeight[gauge] / totalWeight;
+        uint32 currentCycle = (uint32(block.timestamp) + gaugeCycleLength) / gaugeCycleLength * gaugeCycleLength; // todo consider cycle as input parameter
```

Comment:
Why are we dividing and multiplying by the same number? Doesn't that just get us back to the original number from the expression `uint32(block.timestamp) + gaugeCycleLength)`?